pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -257,48 +275,276 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 
 	ctx := context.Background()
 
-	network, err := installation.GetInstallation(context.Background(), r.client, r.provider)
+	ls, err := GetLogStorage(ctx, r.client)
+	if err != nil {
+		// Not finding the LogStorage CR is not an error, as a Managed cluster will not have this CR available but
+		// there are still "LogStorage" related items that need to be set up
+		if !errors.IsNotFound(err) {
+			r.status.SetDegraded("An error occurred while trying to retrieve the LogStorage CR", err.Error())
+			return reconcile.Result{}, err
+		}
+		r.status.OnCRNotFound()
+	} else {
+		r.status.OnCRFound()
+	}
+
+	installationCR, err := installation.GetInstallation(context.Background(), r.client, r.provider)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			r.status.SetDegraded("Installation not found", err.Error())
 			return reconcile.Result{}, err
 		}
-		r.status.SetDegraded("Error querying installation", err.Error())
+		r.status.SetDegraded("An error occurred while trying to retrieve the Installation CR", err.Error())
 		return reconcile.Result{}, err
 	}
 
-	if network.Status.Variant != operatorv1.TigeraSecureEnterprise {
+	// These checks ensure that we're in the correct state to continue to the render function without causing a panic
+	if installationCR.Status.Variant != operatorv1.TigeraSecureEnterprise {
 		r.status.SetDegraded(fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise), "")
 		return reconcile.Result{}, nil
+	} else if ls == nil && installationCR.Spec.ClusterManagementType != operatorv1.ClusterManagementTypeManaged {
+		err := fmt.Errorf("LogStorage must exist for '%s' cluster type", installationCR.Spec.ClusterManagementType)
```

Comment:
Like we discussed I think this is confusing since in this branch the TigeraStatus object won't exist (due to the OnCRNotFound call above). 

Not a problem to call SetDegraded, per-se, but also doesn't add any value, so might be less confusing with a comment and/or removing the SetDegraded call.